libbeat/processors/add_process_metadata/config.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_process_metadata
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type config struct {
+	// IgnoreMissing: Ignore errors if event has no PID field.
+	IgnoreMissing bool `config:"ignore_missing"`
+
+	// OverwriteKeys allow target_fields to overwrite existing fields.
+	OverwriteKeys bool `config:"overwrite_keys"`
+
+	// RestrictedFields make restricted fields available (i.e. env).
+	RestrictedFields bool `config:"restricted_fields"`
+
+	// MatchPIDs fields containing the PID to lookup.
+	MatchPIDs []string `config:"match_pids" validate:"required"`
+
+	// Target is the destination root where fields will be added.
+	Target string `config:"target"`
+
+	// Fields is the list of fields to add to target.
+	Fields []string `config:"include_fields"`
+}
+
+// available fields by default
+var defaultFields = common.MapStr{
+	"process": common.MapStr{
+		"name":       nil,
+		"title":      nil,
+		"exe":        nil,
+		"args":       nil,
+		"pid":        nil,
+		"ppid":       nil,
+		"start_time": nil,
+	},
+}
+
+// fields declared in here (and not in validFields) will only appear in an
+// event if requested explicitly (restricted_fields: true).
+var restrictedFields = common.MapStr{
```

Comment:
@adriansr This list here got me confused for my comment above that it's more then `env.  I assume `restrictedFields` would only contain the restricted fields and not `defaultFields + restrictedFields`.

I wonder if you could use `MapStr.DeepUpdate` to make this happen.

Reply:
have a look now, is this what you expect?